tests/unit/s2n_server_psk_extension_test.c

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#include "crypto/s2n_hmac.h"
+#include "tls/extensions/s2n_server_psk.h"
+
+/* Include source to test static methods. */
+#include "tls/extensions/s2n_server_psk.c"
+
+struct s2n_psk_test_case {
+    s2n_hmac_algorithm hmac_alg;
+    uint8_t hash_size;
+    const uint8_t* identity;
+    size_t identity_size;
+};
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    uint8_t test_identity_1[] = "test identity";
+    uint8_t test_identity_2[] = "another identity";
+
+    struct s2n_psk_test_case test_cases[] = {
+        { .hmac_alg = S2N_HMAC_SHA224, .hash_size = SHA224_DIGEST_LENGTH,
+          .identity = test_identity_1, .identity_size = sizeof(test_identity_1)},
+        { .hmac_alg = S2N_HMAC_SHA384, .hash_size = SHA384_DIGEST_LENGTH,
+          .identity = test_identity_2, .identity_size = sizeof(test_identity_2)},
+    };
```

Comment:
I'm not sure this is a useful set of test cases. You'll always have the same list on both server and client side. You'll want more variation than that. Also, your extension doesn't care about hash algorithm.

Reply:
Will add more unit tests for different list sent on the server and client and will get rid of the hash intialization.